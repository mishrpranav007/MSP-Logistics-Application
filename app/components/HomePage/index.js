import React from 'react';
import {
  StyleSheet,
  View,
  ImageBackground,
  Text,
  Image,
  TouchableOpacity
} from 'react-native';
import PropTypes from 'prop-types';
import homepage from 'app/assets/images/homepage.png';
import logo from 'app/assets/images/imgMspLogo.png';
import scanimage from 'app/assets/images/illoScanBag.png';
import bitmapimage from 'app/assets/images/bitmapImage.png';
import Style from '../../style/index';
const HomePage = ({ navigation }) => (
  <ImageBackground
    testID="home-page"
    source={homepage}
    style={styles.backgroundContainer}
  >
    <View style={styles.logoView}>
      <Image source={logo} style={styles.logoContainer}></Image>
    </View>
    <View style={styles.scanImageContainer}>
      <Image source={scanimage}></Image>
    </View>

    <View>
      <TouchableOpacity
        style={styles.deliveryButtonLogin}
        onPress={() => navigation.navigate('HomeScreen')}
      >
        <View testID="home-page-view" style={styles.bitImageContainer}>
          <Image source={bitmapimage} style={styles.bitmapImageIcon}></Image>
          <Text style={styles.text}>Scan for delivery</Text>
        </View>
      </TouchableOpacity>

      <TouchableOpacity
        style={styles.pickupButtonLogin}
        onPress={() => navigation.navigate('HomeScreen')}
      >
        <View testID="home-page-view" style={styles.bitImageContainer}>
          <Image source={bitmapimage} style={styles.bitmapImageIcon}></Image>
          <Text style={styles.text}>Scan for pickup</Text>
        </View>
      </TouchableOpacity>
    </View>
  </ImageBackground>
);
const styles = StyleSheet.create({
  backgroundContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column'
  },

  logoContainer: {
    marginTop: Style.em(0.5),
    marginLeft: Style.em(5.75),
    marginRight: Style.em(5.5),
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginBottom: Style.em(5.5)
  },
  scanImageContainer: {
    width: Style.em(19.31),
    height: Style.em(15.94),
    marginBottom: Style.em(0.5),
    marginTop: Style.em(0.5)
  },
  deliveryButtonLogin: {
    width: Style.em(18.5),
    height: Style.em(3.25),
    marginTop: Style.em(1.87),
    justifyContent: 'center',
    backgroundColor: Style.PRIMARY_COLOR,
    borderRadius: Style.em(0.69),
    borderTopColor: Style.SECONDARY_COLOR
  },
  pickupButtonLogin: {
    width: Style.em(18.5),
    height: Style.em(3.25),
    justifyContent: 'center',
    backgroundColor: Style.PRIMARY_COLOR,
    marginTop: Style.em(1.25),
    borderRadius: Style.em(0.69),
    marginBottom: Style.em(0.75),
    borderTopColor: Style.SECONDARY_COLOR,
    zIndex: 100
  },
  text: {
    color: Style.WHITE_COLOR,
    fontSize: Style.em(1),
    textAlign: 'center'
  },
  bitImageContainer: {
    flexDirection: 'row',
    justifyContent: 'center'
  },
  bitmapImageIcon: {
    width: Style.em(1.25),
    height: Style.em(1.15),
    marginRight: Style.em(0.62),
    marginBottom: Style.em(0.13),
    opacity: 1
  }
});
HomePage.propTypes = {
  navigation: PropTypes.object
};
export default HomePage;
