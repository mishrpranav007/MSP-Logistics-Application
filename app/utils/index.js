import { Platform, Dimensions } from 'react-native';

export const isAndroid = () => Platform.OS !== 'ios';
export const getScreenWidth = () => Dimensions.get('window').width;
export const getScreenHeight = () => Dimensions.get('window').height;

export const getPlatformString = () => (isAndroid() ? 'android' : 'ios');
export const getFontFamily = font =>
  isAndroid() ? `Futura-${font}` : `FuturaStd-${font}`;

export const isScreenSmall = () => getScreenHeight() <= 550;
