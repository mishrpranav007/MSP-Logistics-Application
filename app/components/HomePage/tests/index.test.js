import React from 'react';
import { renderWithIntl } from 'app/utils/testUtils';
import HomePage from '../index';

describe('<HomePage />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<HomePage />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 homepage screen component', () => {
    const { getAllByTestId } = renderWithIntl(<HomePage />);
    expect(getAllByTestId('home-page').length).toBe(1);
  });
  it('should contain 2 button component inside the homepage screen', () => {
    const { getAllByTestId } = renderWithIntl(<HomePage />);
    expect(getAllByTestId('home-page-view').length).toBe(2);
  });
});
