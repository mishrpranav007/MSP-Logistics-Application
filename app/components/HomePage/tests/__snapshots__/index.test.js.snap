// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/images/homepage.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
        testID="home-page"
      />
      <View>
        <Image
          source={
            Object {
              "testUri": "../../../app/assets/images/imgMspLogo.png",
            }
          }
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "flex-start",
              "marginBottom": 138.28571428571428,
              "marginLeft": 144.57142857142856,
              "marginRight": 138.28571428571428,
              "marginTop": 12.571428571428571,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "height": 400.77714285714285,
            "marginBottom": 12.571428571428571,
            "marginTop": 12.571428571428571,
            "width": 485.5085714285714,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../app/assets/images/illoScanBag.png",
            }
          }
        />
      </View>
      <View>
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Object {
              "backgroundColor": "rgb(36,106,116)",
              "borderRadius": 17.348571428571425,
              "borderTopColor": "rgba(19,43,46,0.83)",
              "height": 81.71428571428572,
              "justifyContent": "center",
              "marginTop": 47.01714285714286,
              "width": 465.1428571428571,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
            testID="home-page-view"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../app/assets/images/bitmapImage.png",
                }
              }
              style={
                Object {
                  "height": 28.91428571428571,
                  "marginBottom": 3.2685714285714287,
                  "marginRight": 15.588571428571427,
                  "opacity": 1,
                  "width": 31.428571428571427,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": 25.142857142857142,
                  "textAlign": "center",
                }
              }
            >
              Scan for delivery
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Object {
              "backgroundColor": "rgb(36,106,116)",
              "borderRadius": 17.348571428571425,
              "borderTopColor": "rgba(19,43,46,0.83)",
              "height": 81.71428571428572,
              "justifyContent": "center",
              "marginBottom": 18.857142857142858,
              "marginTop": 31.428571428571427,
              "width": 465.1428571428571,
              "zIndex": 100,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
            testID="home-page-view"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../app/assets/images/bitmapImage.png",
                }
              }
              style={
                Object {
                  "height": 28.91428571428571,
                  "marginBottom": 3.2685714285714287,
                  "marginRight": 15.588571428571427,
                  "opacity": 1,
                  "width": 31.428571428571427,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": 25.142857142857142,
                  "textAlign": "center",
                }
              }
            >
              Scan for pickup
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;
